### PR TITLE
kafka(ticdc): add unit test to cover claim-check functionality and use uuid to generate object key

### LINKED ISSUE #9528: canal-json add unit test to cover claim-check functionality

### LINKED ISSUE #9529: open-protocol add unit test to cover claim-check functionality

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #9528, close #9529

### What is changed and how it works?

* use `uuid` to generate object key for claim check large message
* add e2e unit test to cover claim-check functionality for both open-protocol and canal-json

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`
```

### CODE REVIEW

**Review by nongfushanquan — APPROVED** [review]

File: pkg/sink/codec/common/message.go [hunk 1]
```diff
@@ -172,30 +170,15 @@ func UnmarshalClaimCheckMessage(data []byte) (*ClaimCheckMessage, error) {
...
-	elements = append(elements, e.GetHandleKeyColumnValues()...)
-	fileName := strings.Join(elements, "-")
-	fileName += ".json"
-
-	fileName = path.Join(date, fileName)
+// NewClaimCheckFileName return the file name for the message which is delivered to the external storage system.
+// UUID V4 is used to generate random and unique file names.
+// This should not exceed the S3 object name length limit.
+// ref https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html
+func NewClaimCheckFileName() string {
```
Comment (CharlesCheung96) [inline pkg/sink/codec/common/message.go:177]: Should NewClaimCheckFileName be moved into claimcheck pkg?

File: pkg/sink/codec/common/message.go [hunk 2]
```diff
@@ -172,30 +170,15 @@ func UnmarshalClaimCheckMessage(data []byte) (*ClaimCheckMessage, error) {
 	return &m, err
 }
 
-// NewClaimCheckFileName return file name for sent the message to claim check storage.
-// make sure the file name can identify one event uniquely.
-// {date}/{schema}-{table}-{commitTs}-{startTs}-{handleKeys}.json
-// the files is organized by date, it's easy to clean up those objects by removing the whole directory.
-func NewClaimCheckFileName(e *model.RowChangedEvent) string {
```
Comment (CharlesCheung96) [inline pkg/sink/codec/common/message.go:179]: Perhaps this logic can be retained to meet the possible needs of users in future?

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**